src/Cortside.AspNetCore/Builder/WebApiBuilder.cs

```diff
@@ -29,6 +29,19 @@ public class WebApiBuilder {
         private bool executingIsEntryAssembly;
         private string url;
 
+        private Action<IHostBuilder> hostConfigurationAction = null;
+        private Action<IWebHostBuilder> webHostConfigurationAction = null;
+
+        public WebApiBuilder WithHostAction(Action<IHostBuilder> action) {
+            this.hostConfigurationAction = action;
+            return this;
+        }
+
+        public WebApiBuilder WithWebHostAction(Action<IWebHostBuilder> action) {
```

Comment:
I like this approach, it matches with a similar method on WebApplicationFactory now.  Can I have you rename the methods to end with Builder instead of Action?

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1.withwebhostbuilder?view=aspnetcore-8.0#microsoft-aspnetcore-mvc-testing-webapplicationfactory-1-withwebhostbuilder(system-action((microsoft-aspnetcore-hosting-iwebhostbuilder)))